enhancements/network/ingress-logging-api.md

```diff
@@ -0,0 +1,364 @@
+	//
+	// * Syslog
+	//
+	// Access logs are sent to a syslog endpoint.  The administrator must
+	// specify an endpoint that can receive syslog messages.  The
+	// expectation is that the administrator has configured a custom syslog
+	// instance.
+	//
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	// +required
+	Type LoggingDestinationType `json:"type"`
+
+	// syslog holds parameters for a syslog endpoint.  Present only if
+	// type is Syslog.
+	//
+	// +optional
+	Syslog *SyslogLoggingDestinationParameters `json:"syslog,omitempty"`
+
+	// container holds parameters for the Container logging destination.
+	// Present only if type is Container.
+	//
+	// +optional
+	Container *ContainerLoggingDestinationParameters `json:"container,omitempty"`
+}
+```
+
+The union discriminator can have either the value `Container` or the value
+`Syslog`:
+
+```go
+// LoggingDestinationType is a type of destination to which to send log
+// messages.
+//
+// +kubebuilder:validation:Enum=Container;Syslog
+type LoggingDestinationType string
+
+const (
+	// Container sends log messages to a sidecar container.
+	ContainerLoggingDestinationType LoggingDestinationType = "Container"
+
+	// Syslog sends log messages to a syslog endpoint.
+	SyslogLoggingDestinationType LoggingDestinationType = "Syslog"
+)
+```
+
+For `Syslog`, an endpoint must be specified, and a facility may be specified (if
+unspecified the default is "local1"); `Container` has no parameters:
+
+```go
+// SyslogLoggingDestinationParameters describes parameters for the Syslog
+// logging destination type.
+type SyslogLoggingDestinationParameters struct {
+	// endpoint identifies the syslog host:port endpoint that receives log
+	// messages.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Endpoint string `json:"endpoint"`
+
+	// facility specifies the syslog facility of log messages.
+	//
+	// If this field is empty, the facility is "local1".
+	//
+	// +optional
+	Facility string `json:"facility,omitempty"`
+}
+
+// ContainerLoggingDestinationParameters describes parameters for the Container
+// logging destination type.
+type ContainerLoggingDestinationParameters struct {
+}
+```
+
+To disable access logging, the user leaves `spec.logging` or
+`spec.logging.access` unspecified.  Otherwise the user must specify a
+destination using `spec.logging.access.destination`.  To specify a destination,
+the user must specify either `Container` or `Syslog` for
+`spec.logging.access.destination.type`.  If the destination type is `Syslog`,
+the user must specify a destination endpoint using
+`spec.logging.access.destination.syslog.endpoint` and may specify a facility
+using `spec.logging.access.destination.syslog.facility`.  The user may specify
+`spec.logging.access.httpLogFormat` to customize the log format.  For example, the following is the definition of an IngressController that logs to a syslog endpoint with IP address 1.2.3.4 and port 10514:
```

Comment:
Thinking of how we're dealing with validation and admission-condition with other fields — is there any opportunity here for us to do defaulting or validation during admission to make this as easy as possible to understand without requiring the user to do separate update/verify steps to understand whether they have a valid config or what their effective config is?

Reply:
>Do you mean defaulting as far as setting empty fields explicitly to their default values during admission?

Yes

> Is there anything we can do now, or is this just something to keep in mind when we add an admission controller?

Maybe let's figure that out — otherwise are we going to have to add another status field?

Reply:
I feel like status is unnecessary because there is little room for varying interpretation—if you specify `spec.logging.access`, then access logging is enabled, the log format is used verbatim, the syslog facility is strictly validated, and the endpoint is used verbatim.  The only thing I can think of is that the operator could probe the syslog endpoint and report a status condition; would that be useful?

Reply:
I agree with your status assessment.

Re: probing, that could be useful, but probably out of scope for the initial feature. What do you think?